cloud/vmware/vmware_vm_shell.py

```diff
@@ -156,18 +141,33 @@ def main():
     if not HAS_PYVMOMI:
         module.fail_json(msg='pyvmomi is required for this module')
 
+    datacenter_name = p['datacenter']
+    cluster_name = p['cluster']
+
     try:
         p = module.params
         content = connect_to_api(module)
 
-        vm = find_vm(content, p['vm_id'], p['vm_id_type'], p['datacenter'])
+        datacenter = None
+        if datacenter_name:
+            datacenter = find_datacenter_by_name(content, datacenter_name)
+            if not datacenter:
+                module.fail_json(msg="datacenter not found")
+
+        cluster = None
+        if cluster_name:
+            cluster = find_cluster_by_name(content, cluster_name, datacenter)
+            if not cluster:
+                module.fail_json(msg="cluster not found")
+
+        vm = find_vm_by_id(content, p['vm_id'], p['vm_id_type'], datacenter, cluster)
         if not vm:
-            module.fail_json(msg='failed to find VM')
+            module.fail_json(msg='VM not found')
 
         msg = execute_command(content, vm, p['vm_username'], p['vm_password'], 
                               p['vm_shell'], p['vm_shell_args'], p['vm_shell_env'], p['vm_shell_cwd'])
 
-        module.exit_json(changed=False, virtual_machines=vm.name, msg=msg)
+        module.exit_json(changed=False, uuid=vm.summary.config.uuid, msg=msg)
```

Comment:
Hi @ritzk - I might overlooked this in the original PR...  Did we want to exit here with changed=False?